docs/windows.md

```diff
@@ -1,142 +1,94 @@
-Windows has a 'git' binary in a 'Git\bin' directory and another one in
-a 'Git\cmd' directory. You should only put the 'Git\cmd' directory in
-the Path environment variable.
-
 ## Install Go
 
-Please install the Go language as explained on
-https://golang.org/doc/install.
-
-To properly install Go, you will need to set some environment
-variables. We recommend you to set them globally using the Control
-Panel, as explained in the documentation above, so that these
-environment variables are automatically available in all the possible
-environments that you might want to use like Git Bash, Windows's cmd,
-Cygwin's terminal, Windows' PowerShell and so on.
-
-You must make sure that the GOROOT environment variable is set and
-that the %GOROOT%/bin directory is in the Path environment variable.
-
-The GOPATH environment variable should also be set to a directory that
-you have created, and the %GOPATH/bin directory should also be in the
-Path environment variable.
-
-## Download go-ipfs and fix Git authentication
+`go-ipfs` is built on Golang and thus depends on it for all building methods.  
+https://golang.org/doc/install  
+The `GOPATH` environment variable must be set as well.  
+https://golang.org/doc/code.html#GOPATH
 
-Use the following command to download go-ipfs source code:
-
-```
+## Choose the way you want to proceed
+`go-ipfs` utilizes `make` to automate builds and testing, but can be built without it using only `git` and `go`.  
+The `make` portions cover setting up the environment, "minimal" assumes you have a working version of `go` and `git` already.  
+[make→](#make)  
+[minimal→](#minimal)
+
+## Make
+There are currently 2 documented ways of building with `make` on Windows.  
+[MSYS2→](#msys2)  
+[Cygwin→](#cygwin)
+
+## MSYS2
+Install msys2 (http://www.msys2.org)  
+Run the following inside a normal `cmd` prompt (Not the MSYS2 prompt, we only need MSYS2's tools).  
+An explanation is below.
+```
+SET PATH=%PATH%;\msys64\usr\bin
+pacman --noconfirm -S  git make unzip
 go get -u github.com/ipfs/go-ipfs
+cd %GOPATH%\src\github.com\ipfs\go-ipfs
+make install
 ```
 
-The above command uses Git to download go-ipfs from its GitHub
-repository. If you get authentication problems with Git, you might
-want to take a look at
-https://help.github.com/articles/caching-your-github-password-in-git/
-and use the suggested solution:
+|      |      |
+| ---: | :--- |
+|`SET PATH=%PATH%;\msys64\usr\bin`         |Add msys2's bin to `PATH`; Defaults to: (\msys64\usr\bin)|
+|`pacman --noconfirm -S  git make unzip`   |Install `go-ipfs` build dependencies|
+|`go get -u github.com/ipfs/go-ipfs`       |Fetch / Update `go-ipfs` source|
+|`cd %GOPATH%\src\github.com\ipfs\go-ipfs` |Change to `go-ipfs` directory|
+|`make install`                            |Build and install to `%GOPATH%\bin\ipfs.exe`|
 
+You can permenantly append the MSYS2 bin to your `PATH` so that on subsequent builds you can simply run
```

Comment:
>From my recollection of doing Windows dev years ago, this was non-trivial and buried in system settings somewhere

The biggest problem is that it's different across versions, even on Windows 10 alone I've noticed they changed the interface 3 times now.

There's the `setx` command but I need to test access and availability of it. I think users can invoke it normally and I'm seeing Windows 7 as the minimum version but it may only be included with a specific SDK until Windows 8. If it's included in stock Windows 7 I'd be comfortable just recommending that as the means. The only problem I can think of is that it would be bad if users made a typo for their `PATH`
i.e
`setx PATH %PATH;\some\other\path`
notice the lack of the closing % which would just ruin everything. I guess that can't really be avoided either way though.

I'll write a revision including this and maybe a notice "be very careful to check your input here..."

Reply:
> The biggest problem is that it's different across versions, even on Windows 10 alone I've noticed they changed the interface 3 times now.

Is this a good resource to link to? https://www.computerhope.com/issues/ch000549.htm